lddwrap/__init__.py

```diff
@@ -0,0 +1,228 @@
+                 soname: Optional[str],
+                 path: Optional[pathlib.Path],
+                 found: bool,
+                 mem_address: Optional[str],
+                 unused: Optional[bool] = None) -> None:
+        """
+        Initialise dependency.
+
+        :param soname: library name
+        :param path: path of the library
+        :param found: library found status
+        :param mem_address: hex memory location
+        :param unused: library not used
+        """
+        # pylint: disable= too-many-arguments
+        self.soname = soname
+        self.path = path
+        self.found = found
+        self.mem_address = mem_address
+        self.unused = unused
+
+    def __str__(self):
+        """Print dependency information."""
+        return "soname: {}, path: {}, found: {}, mem_address: {}, unused: {}".format(
+            self.soname, self.path, self.found, self.mem_address, self.unused)
+
+
+def parse_line(line: str) -> Dependency:
+    """
+    Parse single line of ldd output.
+
+    :param line: to parse
+    :return: dependency or none if line was empty
+
+    """
+    found = not 'not found' in line
+    parts = [part.strip() for part in line.split(' ')]
+
+    # There are two types of outputs for a dependency, with or without soname. For example:
+    # with soname: 'libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9a19d8a000)'
+    # without soname: '/lib64/ld-linux-x86-64.so.2 (0x00007f9a1a329000)'
+    # with soname but not found: 'libboost_program_options.so.1.62.0 => not found'
+    # with soname but without rpath: 'linux-vdso.so.1 =>  (0x00007ffd7c7fd000)'
+    if '=>' in line:
+        if len(parts) != 4:
+            raise RuntimeError("Expected 4 parts in the line but found {}: {}".format(len(parts), line))
+
+        soname = None
+        dep_path = None
+        mem_address = None
+        if found:
+            soname = parts[0]
+            if parts[2] != '':
+                dep_path = pathlib.Path(parts[2])
+            mem_address = parts[3]
+        else:
+            if "/" in parts[0]:
+                dep_path = pathlib.Path(parts[0])
+            else:
+                soname = parts[0]
+
+        return Dependency(soname=soname, path=dep_path, found=found, mem_address=mem_address)
+    else:
+        if len(parts) != 2:
+            raise RuntimeError("Expected 2 parts in the line but found {}: {}".format(len(parts), line))
+
+        return Dependency(soname=None, path=pathlib.Path(parts[0]), found=True, mem_address=parts[1])
+
+
+def get_dependencies(path: pathlib.Path,
+                     verbose: bool = False,
+                     unused: bool = False,
+                     env: Optional[Dict[str, str]] = None) -> List[Dependency]:
+    """
+    Retrieve a list of dependencies of the given binary.
+
+    :param path: path to a file
+    :param verbose: additionally print all information
+    :param unused: additional adds information if dependency is used
```

Comment:
Comment B:

I think it would be easier to understand the descriptions if you wrote something like this:

```
:param verbose if set, print all the information
:param unused: if set, check if dependencies are used
```

Could you please take a look at all the descriptions and fix if necessary?